File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.py
class Solution:
    def sortVowels(self, s: str) -> str:
        vowels = ["A", "E", "I", "O", "U", "a", "e", "i", "o", "u"]
        
        s_vowels = []
        for c in s:
            if c not in vowels:
                continue
            s_vowels.append(c)
            
        s_vowels.sort()
        result = ""
        
        idx = 0
        for c in s:
            if c not in vowels:
                result += c
            else:
                result += s_vowels[idx]
                idx += 1
                
        if idx < len(s_vowels):
            while idx < len(s_vowels):
                result += s_vowels[idx]
                idx += 1
                
        return result
        
        
                
        

File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var sortVowels = function(s) {
    let vowels = ['a','e','i','o','u','A','E','I','O','U'];
    let sv = [];
    
    for(let i = 0; i < s.length; i++){
        if(vowels.includes(s[i])){
            sv.push(s[i]);
        }
    }
    
    sv.sort();
    let result = "";
    let idx = 0;
    
    for(let i = 0; i < s.length; i++){
        if(vowels.includes(s[i])){
            result += sv[idx];
            idx += 1;
        }
        else{
            result += s[i];
        }
    }
    
    return result;
};